src/averageAges.js

```diff
@@ -53,6 +87,29 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+
+  const mothers = people.filter(person =>
+    people.some(child => child.name === person.mother));
+
+  const mothersWithSon = people.filter(person =>
+    people.some(child => child.name === person.mother
+      && person.sex === 'm'));
+
+  const ageDiffMothersAllkids = mothers.map(person =>
+    person.born - people.find(woman => woman.name === person.mother).born);
+
+  const ageDiffMothersSons = mothersWithSon.map(person =>
+    person.born - people.find(woman => woman.name === person.mother).born);
+
+  const sumAgeDiff = arguments.length < 2
```

Comment:
```suggestion
  const sumAgeDiff = onlySons
```
create variable for this